[PUB-3498] Add post component

anafilipadealmeida opened the pull request:
<!--- Provide a general summary of your changes in the Title above. -->

## Description
- Add post component for the calendar week view, with a `small` variation for the month view
- Create ChannelIcon shared component, as it might be useful for other things in the future

To do later, in a different PR: add strings to the en_us.json file, replace static data

## Context & Notes
https://buffer.atlassian.net/browse/PUB-3498
One of the accessibility props will work once we deploy this PR I created in the ui repo, to add titles to svg icons when they are meaningful:  https://github.com/bufferapp/ui/pull/297
Tagged Lauren on the "reminder" bell icon, as adding a round background myself in css will always look weird 😅 

<!--- Why is this change required? What problem does it solve? -->
<!--- Is there a related JIRA card? Please link to it here. -->

## Screenshots (if appropriate)
<img width="418" alt="Screenshot 2021-05-03 at 15 31 56" src="https://user-images.githubusercontent.com/16758464/116889760-d8247480-ac24-11eb-98c9-e11425d54bd6.png">
<img width="415" alt="Screenshot 2021-05-03 at 15 32 02" src="https://user-images.githubusercontent.com/16758464/116889766-d955a180-ac24-11eb-83c0-571b1626527f.png">
<img width="410" alt="Screenshot 2021-05-03 at 15 32 06" src="https://user-images.githubusercontent.com/16758464/116889768-d955a180-ac24-11eb-92e6-c42c5ab4e99f.png">
<img width="425" alt="Screenshot 2021-05-03 at 15 24 25" src="https://user-images.githubusercontent.com/16758464/116888989-fdfd4980-ac23-11eb-8d67-518d8a7d59c9.png">
<img width="422" alt="Screenshot 2021-05-03 at 15 24 33" src="https://user-images.githubusercontent.com/16758464/116888992-ff2e7680-ac23-11eb-8cf0-d1ee8112bf0d.png">
<img width="415" alt="Screenshot 2021-05-03 at 15 30 16" src="https://user-images.githubusercontent.com/16758464/116889521-91cf1580-ac24-11eb-8497-7804d4951450.png">
<img width="415" alt="Screenshot 2021-05-03 at 15 30 23" src="https://user-images.githubusercontent.com/16758464/116889524-9267ac00-ac24-11eb-8f00-874a9eb1150e.png">
<img width="425" alt="Screenshot 2021-05-03 at 15 31 45" src="https://user-images.githubusercontent.com/16758464/116889710-cba01c00-ac24-11eb-9aed-23260fa7f4e0.png">
<img width="424" alt="Screenshot 2021-05-03 at 15 24 49" src="https://user-images.githubusercontent.com/16758464/116889152-271dda00-ac24-11eb-967b-7088c6c7b661.png">
<img width="464" alt="Screenshot 2021-05-03 at 15 25 10" src="https://user-images.githubusercontent.com/16758464/116889157-27b67080-ac24-11eb-8b74-8bc9e9aed1b9.png">
<img width="414" alt="Screenshot 2021-05-03 at 15 29 09" src="https://user-images.githubusercontent.com/16758464/116889361-5d5b5980-ac24-11eb-9ab3-6eaf97b828fd.png">
<img width="413" alt="Screenshot 2021-05-03 at 15 29 16" src="https://user-images.githubusercontent.com/16758464/116889367-5e8c8680-ac24-11eb-8d1a-df1e77254b3c.png">




## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

-   [x] My code follows the code style and guidelines of this project. <!--- eslint -->
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. <!--- e.g. team members, different plans -->
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

anafilipadealmeida commented on packages/calendar/components/PostItem/index.jsx:
```diff
@@ -0,0 +1,116 @@
+};
+
+PostItem.propTypes = {
+  small: PropTypes.bool,
```
@hamstu, I took this structure from the api planning paper doc, but this can all be replaced by a different structure once there's a solid plan!

hamstu replied:
This is fantastic @anafilipadealmeida! Yeah let's chat more on the API today.

hamstu commented:
@anafilipadealmeida Small note to note forget the `box-shadow` on these. (The figma seems to suggest `filter: drop-shadow(...)` but I'm sure a box-shadow would work as long as it looks like a good match.

albennett commented on packages/calendar/components/PostItem/index.jsx:
```diff
@@ -0,0 +1,116 @@
+  return (
+    <PostWrapper
+      small={small}
+      campaignColor={post.campaign.color}
```
Will the graphql response include an empty object for the campaign if a post isn't associated with it? If not, we may want to add a `post.campaign?.color`

albennett commented:
This is looking great @anafilipadealmeida . I love how you've kept accessibility in mind. I do think the reminders bell looks a bit off. If it's hard to make the icon smaller, I think it's best without the circle background 🤔 . I know you're in talks with Lauren about this in jira! 😄 

I tried running storybook to look at the different states but the build was broken with an app-shell error with the Field component. Did you run into this?

anafilipadealmeida commented:
> This is looking great @anafilipadealmeida . I love how you've kept accessibility in mind. I do think the reminders bell looks a bit off. If it's hard to make the icon smaller, I think it's best without the circle background 🤔 . I know you're in talks with Lauren about this in jira! 😄
> 
> I tried running storybook to look at the different states but the build was broken with an app-shell error with the Field component. Did you run into this?

Yes @albennett , I'll make some small ui adjustments based on the chat with Lauren! Great reminder about storybook, I wanted to follow up on the issue - I had to go to node_modules and capitalize that file name to make our own storybook to run, which means there's something off in the app-shell repo.

letsbleachthis commented:
Something else just came to mind! @anafilipadealmeida are these currently using a fixed width, or do they have min and max values? It would be awesome if they could shrink and grow (within some set parameters) to adjust to various screen sizes.

anafilipadealmeida commented:
Thank you all for the comments ❤️ I only had time to revisit this now! 
@letsbleachthis, yes, the post items are quite flexible and follow the width of the parent, so they have the ability to shrink and grow. I've also removed the red border from the error posts as you asked in Jira. I also added some tweaks to get the bell icon with a nice look.
@hamstu great catch! I added the BDS box-shadow to the postItems, which is the same we have in Posts.

@hamstu or @albennett I'd love to have your approval here, so we can close this one for now!

albennett reviewed (approved):
![](https://media0.giphy.com/media/oGO1MPNUVbbk4/giphy.gif)

hamstu commented:
Awesome stuff @anafilipadealmeida! This is a small one but I think the bottom padding on the icon(s) is a but smaller than it should be. This is in the figma: 
![image](https://user-images.githubusercontent.com/809093/116934033-4105ef80-ac19-11eb-92cc-e0bcbfcaaad8.png)
